libtock/util/streaming_process_slice.c

```diff
@@ -101,43 +94,55 @@ returncode_t streaming_process_slice_get_and_swap(
 
 returncode_t streaming_process_slice_deinit(
   streaming_process_slice_state_t* state,
-  uint8_t**                        buffer,
-  size_t*                          size) {
-  uint8_t* ret_buffer;
-  size_t ret_size;
+  uint8_t**                        buffer_a,
+  size_t*                          size_a,
+  uint8_t**                        buffer_b,
+  size_t*                          size_b) {
+  uint8_t* ret_buffer_a, * ret_buffer_b;
+  size_t ret_size_a, ret_size_b;
 
   // Unallow the buffer currently allowed to the kernel:
   allow_rw_return_t unallow_res =
     allow_readwrite(state->driver, state->allow, NULL, 0);
 
   if (unallow_res.success) {
     // Unallow failed, don't modify the state struct.
-    ret_buffer = NULL;
-    ret_size   = 0;
+    ret_buffer_a = NULL;
+    ret_size_a   = 0;
+    ret_buffer_b = NULL;
+    ret_size_b   = 0;
```

Comment:
Silly comment, but is this logic right? `unallow_res.success` being `true` means the unallow failed?

Reply:
Yikes!

Reply:
Fixed.